### PR TITLE
Fix exception handling in DrainStatusPoller

### LINKED ISSUE #9143: DrainStatusPoller KeyError crash in 1.1.18
I've just seen this reported in wmstats for vocms0282. Traceback is in [1]

[1]
```
2019-04-06 20:00:16,208:140226378852096:INFO:DrainStatusPoller:Running agent drain algorithm...
2019-04-06 20:00:16,461:140226387244800:WARNING:ResourceControlUpdater:Site T2_TH_CUNSTDA has invalid IO thresholds in SSB. Taking no action
2019-04-06 20:00:16,462:140226387244800:WARNING:ResourceControlUpdater:Site: T3_US_Omaha has incomplete SSB metrics, see {'slotsCPU': 2001}
2019-04-06 20:00:16,462:140226387244800:WARNING:ResourceControlUpdater:Site: T2_TH_CUNSTDA has incomplete SSB metrics, see {'slotsCPU': 67}
2019-04-06 20:00:16,519:140226387244800:ERROR:GenericDataCache:Traceback (most recent call last):
  File "/data/srv/wmagent/v1.1.20.patch3/sw/slc7_amd64_gcc630/cms/wmagent/1.1.20.patch3/lib/python2.7/site-packages/WMCore/Cache/GenericDataCache.py", line 37, in getData
    self.data = self.func(**self.kwargs)
  File "/data/srv/wmagent/v1.1.20.patch3/sw/slc7_amd64_gcc630/cms/wmagent/1.1.20.patch3/lib/python2.7/site-packages/WMCore/Services/ReqMgrAux/ReqMgrAux.py", line 49, in _getResult
    verb=verb, contentType=contentType)
  File "/data/srv/wmagent/v1.1.20.patch3/sw/slc7_amd64_gcc630/cms/wmagent/1.1.20.patch3/lib/python2.7/site-packages/WMCore/Services/Service.py", line 211, in refreshCache
    return open(cachefile, 'r')
IOError: [Errno 2] No such file or directory: '/data/srv/wmagent/v1.1.20.patch3/install/wmagent/AgentStatusWatcher/.wmcore_cache/.wmcore_cache_31961/requests/cmsweb.cern.ch:8443/15744094587142094_GET_wmagentconfig_vocms0282.cern.ch'

2019-04-06 20:00:16,520:140226378852096:ERROR:GenericDataCache:Traceback (most recent call last):
  File "/data/srv/wmagent/v1.1.20.patch3/sw/slc7_amd64_gcc630/cms/wmagent/1.1.20.patch3/lib/python2.7/site-packages/WMCore/Cache/GenericDataCache.py", line 37, in getData
    self.data = self.func(**self.kwargs)
  File "/data/srv/wmagent/v1.1.20.patch3/sw/slc7_amd64_gcc630/cms/wmagent/1.1.20.patch3/lib/python2.7/site-packages/WMCore/Services/ReqMgrAux/ReqMgrAux.py", line 49, in _getResult
    verb=verb, contentType=contentType)
  File "/data/srv/wmagent/v1.1.20.patch3/sw/slc7_amd64_gcc630/cms/wmagent/1.1.20.patch3/lib/python2.7/site-packages/WMCore/Services/Service.py", line 211, in refreshCache
    return open(cachefile, 'r')
IOError: [Errno 2] No such file or directory: '/data/srv/wmagent/v1.1.20.patch3/install/wmagent/AgentStatusWatcher/.wmcore_cache/.wmcore_cache_31961/requests/cmsweb.cern.ch:8443/15744094587142094_GET_wmagentconfig_vocms0282.cern.ch'

2019-04-06 20:00:16,520:140226387244800:WARNING:ReqMgrAux:agent config is not correct: {}
2019-04-06 20:00:16,520:140226378852096:WARNING:ReqMgrAux:agent config is not correct: {}
2019-04-06 20:00:16,520:140226387244800:ERROR:ResourceControlUpdater:Error occurred, will retry later:
2019-04-06 20:00:16,521:140226387244800:ERROR:ResourceControlUpdater:'UserDrainMode'
2019-04-06 20:00:16,524:140226387244800:ERROR:ResourceControlUpdater:Trace back: 
Traceback (most recent call last):
  File "/data/srv/wmagent/v1.1.20.patch3/sw/slc7_amd64_gcc630/cms/wmagent/1.1.20.patch3/lib/python2.7/site-packages/WMComponent/AgentStatusWatcher/ResourceControlUpdater.py", line 104, in algorithm
    self.getAgentsByTeam()
  File "/data/srv/wmagent/v1.1.20.patch3/sw/slc7_amd64_gcc630/cms/wmagent/1.1.20.patch3/lib/python2.7/site-packages/WMComponent/AgentStatusWatcher/ResourceControlUpdater.py", line 120, in getAgentsByTeam
    if isDrainMode(self.config):
  File "/data/srv/wmagent/v1.1.20.patch3/sw/slc7_amd64_gcc630/cms/wmagent/1.1.20.patch3/lib/python2.7/site-packages/WMCore/Services/ReqMgrAux/ReqMgrAux.py", line 199, in isDrainMode
    return AUXDB_AGENT_CONFIG_CACHE["UserDrainMode"] or AUXDB_AGENT_CONFIG_CACHE["AgentDrainMode"]
KeyError: 'UserDrainMode'

2019-04-06 20:00:16,524:140226387244800:INFO:ResourceControlUpdater:Resource control cycle finished updating site state and thresholds.
2019-04-06 20:00:16,524:140226387244800:INFO:BaseWorkerThread:ResourceControlUpdater took 0.414 secs to execute
2019-04-06 20:00:16,545:140226378852096:ERROR:BaseWorkerThread:Error in worker algorithm (1):
Backtrace:
  <WMComponent.AgentStatusWatcher.DrainStatusPoller.DrainStatusPoller object at 0x7f89041435d0> 'UserDrainMode'  File "/data/srv/wmagent/v1.1.20.patch3/sw/slc7_amd64_gcc630/cms/wmagent/1.1.20.patch3/lib/python2.7/site-packages/WMCore/WorkerThreads/BaseWorkerThread.py", line 182, in __call__
    tSpent, results, _ = algorithmWithDBExceptionHandler(parameters)
  File "/data/srv/wmagent/v1.1.20.patch3/sw/slc7_amd64_gcc630/cms/wmagent/1.1.20.patch3/lib/python2.7/site-packages/WMCore/Database/DBExceptionHandler.py", line 35, in wrapper
    return f(*args, **kwargs)
  File "/data/srv/wmagent/v1.1.20.patch3/sw/slc7_amd64_gcc630/cms/wmagent/1.1.20.patch3/lib/python2.7/site-packages/Utils/Timers.py", line 24, in wrapper
    res = func(*arg, **kw)
  File "/data/srv/wmagent/v1.1.20.patch3/sw/slc7_amd64_gcc630/cms/wmagent/1.1.20.patch3/lib/python2.7/site-packages/WMComponent/AgentStatusWatcher/DrainStatusPoller.py", line 48, in algorithm
    if isDrainMode(self.config):
  File "/data/srv/wmagent/v1.1.20.patch3/sw/slc7_amd64_gcc630/cms/wmagent/1.1.20.patch3/lib/python2.7/site-packages/WMCore/Services/ReqMgrAux/ReqMgrAux.py", line 199, in isDrainMode
    return AUXDB_AGENT_CONFIG_CACHE["UserDrainMode"] or AUXDB_AGENT_CONFIG_CACHE["AgentDrainMode"]

2019-04-06 20:00:16,545:140226378852096:INFO:Harness:>>>Terminating worker threads
2019-04-06 20:00:16,545:140226378852096:ERROR:BaseWorkerThread:Error in event loop (2): <WMComponent.AgentStatusWatcher.DrainStatusPoller.DrainStatusPoller object at 0x7f89041435d0> 'UserDrainMode'
Backtrace:
  File "/data/srv/wmagent/v1.1.20.patch3/sw/slc7_amd64_gcc630/cms/wmagent/1.1.20.patch3/lib/python2.7/site-packages/WMCore/WorkerThreads/BaseWorkerThread.py", line 208, in __call__
    raise ex

2019-04-06 20:00:16,607:140226378852096:INFO:BaseWorkerThread:Worker thread <WMComponent.AgentStatusWatcher.DrainStatusPoller.DrainStatusPoller object at 0x7f89041435d0> terminated
```

### PR DESCRIPTION
**Pull request by goughes**
Fixes #9143

isDrainMode, which can throw a KeyError, was running outside a try/except block

### CODE REVIEW

File: src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py [hunk 1]
```diff
@@ -45,25 +45,24 @@ def algorithm(self, parameters):
...
-        if isDrainMode(self.config):
-            # check to see if the agent hit any speed drain thresholds
-            thresholdsHit = self.checkSpeedDrainThresholds()
-            if thresholdsHit:
-                logging.info("Updating agent configuration for speed drain...")
-                self.updateAgentSpeedDrainConfig(thresholdsHit)
-            # now collect drain statistics
-            try:
+        try:
+            if isDrainMode(self.config):
```
Comment (amaltaro) [inline src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py]: @goughes Erik, as we quickly discussed in the meeting, `isDrainMode` is either using the cache or making a call to reqmgr_aux db again (as done a couple of lines above - getWMAgentConfig).
I think we can replace this line by what we have here:
https://github.com/dmwm/WMCore/blob/dfca72872e8994c761e33c5289c7bbf74e6604fe/src/python/WMCore/Services/ReqMgrAux/ReqMgrAux.py#L196

any thoughts?

Comment (goughes) [reply src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py]: That should work. The only thing different in ReqMgrAux is they check for the keys first:
https://github.com/dmwm/WMCore/blob/36970485a95fe305fd767458c2e32b897a0885a3/src/python/WMCore/Services/ReqMgrAux/ReqMgrAux.py#L194

But, on the off chance that those keys don't exist, the KeyError will be caught and reported.

**Comment by amaltaro** [comment]
DrainStatusPoller also checks it at L44. But it's Ok to have the try/except block, it won't hurt.

**Comment by amaltaro** [comment]
@goughes Erik, can you please fix those 2 pylint bold warnings and squash your commits? Changes look good.

**Pull request merged**